Add an oracledb_session resource

nsdavidson opened the pull request:
This adds an oracle_session resource similar to the existing resource
for MySQL and MSSQL.  It assumes the sqlplus tool is installed and in
the path of the user InSpec connects as.

Signed-off-by: Nolan Davidson <ndavidson@chef.io>

Commit:
Add an oracle_session resource

This adds an oracle_session resource similar to the existing resource
for MySQL and MSSQL.  It assumes the sqlplus tool is installed and in
the path of the user InSpec connects as.

Signed-off-by: Nolan Davidson <ndavidson@chef.io>

Commit:
Changing oracle_session back to using escaped query.

Signed-off-by: Nolan Davidson <ndavidson@chef.io>

aaronlippold reviewed (commented):
You may want to consider adding:
```def initialize(user = nil, pass = nil, host = localhost, service = nil)``` and `... | sqlplus -s #{@user}/#{@pass}@{@host}/#{@service}")`. This will default the value to localhost but as we found in the postgres resource, it is needed and useful sometime to be able to not force the host value on the backend to just be localhost, especially in the context of a remote test.

aaronlippold reviewed (commented):
And also, don't forget to document that you can pass in the host and session values in your examples, but that they default to 'localhost' and to 'nil'.

aaronlippold reviewed (changes requested):
Sorry, forgot to mark this request changes.

chris-rock commented on lib/resources/oracle_session.rb:
```diff
@@ -0,0 +1,39 @@
+    def query(q)
+      escaped_query = q.gsub(/\\/, '\\\\').gsub(/"/, '\\"').gsub(/\$/, '\\$')
+
+      cmd = inspec.command("echo \"#{escaped_query}\" | sqlplus -s #{@user}/#{@pass}@localhost/#{@service}")
```
we need to be able to parametrize the sqlplus bin. e.g. in a oracle docker container https://store.docker.com/images/oracle-database-enterprise-edition this is located in `/u01/app/oracle/product/12.1.0/dbhome_1/bin/sqlplus`

aaronlippold replied:
If we are going to parametrize, let's parametrize! :) 👍 

```def query(q)
+      escaped_query = q.gsub(/\\/, '\\\\').gsub(/"/, '\\"').gsub(/\$/, '\\$')
+
+      cmd = inspec.command("echo \"#{escaped_query}\" | {@sqlplus_bin} {@sqlplus_opts} #{@user}/#{@pass}@{@host}/#{@service}")
```

chris-rock commented on lib/resources/oracle_session.rb:
```diff
@@ -0,0 +1,39 @@
+      end
+    "
+
+    def initialize(user = nil, pass = nil, service = nil)
```
Do we need to support DB_SID?

chris-rock replied:
Not 100% sure yet, but should we use hash options here like:
```
oracle_session(user: 'my_user', password: 'password')
```

aaronlippold replied:
Yes you need to support DB_SID

I agree, a hash makes sense.

chris-rock reviewed (commented):
This is a great addition to inspec @nsdavidson
Do you think we can have some kind of integration testing? At least we should add some unit testing similar to https://github.com/chef/inspec/blob/master/test/unit/resources/mysql_conf_test.rb

Commit:
Refactor to options hash and add unit tests

Switched the oracle_session resource to take an option hash and allow
for configuring hostname, DB_SID, and sqlplus binary path.

Added unit tests.

Signed-off-by: Nolan Davidson <ndavidson@chef.io>

Commit:
Resolving rubucop issues

Signed-off-by: Nolan Davidson <ndavidson@chef.io>

Commit:
Renaming oracle_session to oracledb_session

Signed-off-by: Nolan Davidson <ndavidson@chef.io>

arlimus reviewed (approved):
This is a great start, thank you @nsdavidson !!

Also kudos to the reviews from @chris-rock and @aaronlippold .
